--- net/minecraft/src/ThreadLoginVerifier.java
+++ net/minecraft/src/ThreadLoginVerifier.java
@@ -17,7 +17,7 @@
 	public void run() {
 		try {
 			String var1 = NetLoginHandler.getServerId(this.loginHandler);
-			URL var2 = new URL("http://www.minecraft.net/game/checkserver.jsp?user=" + URLEncoder.encode(this.loginPacket.username, "UTF-8") + "&serverId=" + URLEncoder.encode(var1, "UTF-8"));
+			URL var2 = new URL("https://session.minecraft.net/game/checkserver.jsp?user=" + URLEncoder.encode(this.loginPacket.username, "UTF-8") + "&serverId=" + URLEncoder.encode(var1, "UTF-8"));
 			BufferedReader var3 = new BufferedReader(new InputStreamReader(var2.openStream()));
 			String var4 = var3.readLine();
 			var3.close();
